subsys/net/lib/nrf_cloud/src/nrf_cloud_download.c

```diff
@@ -314,94 +98,24 @@ static int coap_dl(struct nrf_cloud_download_data *const dl)
 		}
 	}
 #endif
-	/* Reset the received count */
-	dl->coap_rcvd_bytes = 0;
-	dl->resume_offset = 0;
-
-	/* Get the options for the proxy download */
-	ret = nrf_cloud_coap_transport_proxy_dl_opts_get(&cc_opts[ACPT_IDX],
-							 &cc_opts[PRXY_IDX],
-							 dl->host, file_path);
+	ret = nrf_cloud_coap_transport_proxy_dl_uri_get(
+		proxy_uri, ARRAY_SIZE(proxy_uri), dl->host, file_path);
 	if (ret) {
-		LOG_ERR("Failed to set CoAP options, error: %d", ret);
+		LOG_ERR("Failed to get CoAP proxy URL, error: %d", ret);
 		return ret;
 	}
 
-	if (dl->type == NRF_CLOUD_DL_TYPE_FOTA) {
-#if defined(CONFIG_FOTA_DOWNLOAD_EXTERNAL_DL)
-		ret = fota_download_external_start(dl->host, file_path, dl->fota.expected_type,
-						  (size_t)dl->fota.img_sz);
-		if (ret) {
-			LOG_ERR("Failed to start FOTA download, error: %d", ret);
-			return ret;
-		}
-#else
-		return -ENOTSUP;
-#endif
-	}
-
-	return coap_dl_start(dl, 0);
-}
-
-int nrf_cloud_download_coap_offset_resume(const size_t offset)
-{
-	int err = 0;
-
-	k_mutex_lock(&active_dl_mutex, K_FOREVER);
-
-	if ((active_dl.type == NRF_CLOUD_DL_TYPE_FOTA) ||
-	    (active_dl.type == NRF_CLOUD_DL_TYPE_DL_CLIENT)) {
-		active_dl.resume_offset = offset;
-		k_work_schedule(&resume_work, K_SECONDS(1));
-	} else if (active_dl.type == NRF_CLOUD_DL_TYPE_NONE) {
-		LOG_ERR("No active download, cannot resume");
-		err = -ENODEV;
-	} else {
-		LOG_ERR("Invalid download type (%d), cannot resume", active_dl.type);
-		err = -EINVAL;
-	}
-
-	k_mutex_unlock(&active_dl_mutex);
-
-	return err;
-}
-
-static void resume_work_fn(struct k_work *unused)
-{
-	int ret;
-
-	if (active_dl.type == NRF_CLOUD_DL_TYPE_NONE) {
-		LOG_ERR("No active download, cannot resume");
-		return;
-	}
-
-	/* Ensure current request is canceled */
-	coap_client_cancel_requests(&coap_client.cc);
-
-	ret = coap_dl_connect_and_auth();
-	if (ret == 0) {
-		LOG_DBG("Resuming download at offset: %u", active_dl.resume_offset);
-		ret = coap_dl_start(&active_dl, active_dl.resume_offset);
-	}
-
-	/* On failure, send the events required to generate the error/done status */
-	if (ret) {
-		struct downloader_evt evt = {0};
-
-		LOG_ERR("Failed to resume CoAP download");
-
-		/* Send a non-recoverable error event (not ECONN) */
-		evt.id = DOWNLOADER_EVT_ERROR;
-		evt.error = -EIO;
-		(void)coap_dl_event_send(&active_dl, &evt);
+	struct downloader_host_cfg host_cfg = {
```

Comment:
variables should be declared at top of function or if block, not in the middle of a function